base/applications/cmdutils/more/more.c

```diff
@@ -121,11 +285,69 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
 
         /* Got our key, return to caller */
         KeyEvent = ir.Event.KeyEvent;
-    }
-    while ((KeyEvent.wVirtualKeyCode == VK_SHIFT) ||
-           (KeyEvent.wVirtualKeyCode == VK_MENU) ||
-           (KeyEvent.wVirtualKeyCode == VK_CONTROL));
 
+        /* Ctrl key is pressed? */
+        fCtrl = !!(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED));
+
+        /* Ctrl+C or Ctrl+Esc? */
+        if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                      (KeyEvent.wVirtualKeyCode == L'C')))
+        {
+            s_chSubCommand = 0;
+            break;
+        }
+
+        /* Ignore any unsupported keyboard press */
+        if ((KeyEvent.wVirtualKeyCode == VK_SHIFT) ||
+            (KeyEvent.wVirtualKeyCode == VK_MENU)  ||
+            (KeyEvent.wVirtualKeyCode == VK_CONTROL))
+        {
+            continue;
+        }
+
+        /* Ignore any unsupported subcommand */
+        if (s_chSubCommand != L'P' && s_chSubCommand != L'S')
+            break;
+
+        ch = KeyEvent.uChar.UnicodeChar;
+        if (L'0' <= ch && ch <= L'9')
+        {
+            if (nLines == 0 && ch == L'0')
+                continue;
+            nLines *= 10;
+            nLines += ch - L'0';
+            ConStreamWrite(Pager->Screen->Stream, &ch, 1);
+            continue;
+        }
+        if (KeyEvent.wVirtualKeyCode == VK_RETURN)
+        {
+            break;
+        }
+        else if (KeyEvent.wVirtualKeyCode == VK_ESCAPE)
+        {
+            s_chSubCommand = L'-';
+            break;
+        }
```

Comment:
If your intent is to just reset the paging prompt without doing anything else (which seems to be the case with Windows' MORE, only when the current prompt is the "OPTIONS"/help one), then it should be better to just reset `s_chSubCommand` to zero, remain within this PagePrompt function, refresh the displayed prompt and wait for keyboard, without using this sort of "hack" of setting the command to '-' and using the hackish MorePageActionDoNothing stuff below.

Reply:
OK.